packages/SwingSet/src/devices/timer/device-timer.js

```diff
@@ -132,33 +149,28 @@ function makeTimerMap(state = undefined) {
   }
 
   // We don't expect this to be called often, so we don't optimize for it.
-  function remove(targetHandler) {
+  /**
+   *
+   * @param {Waker} target
+   * @returns {bigint[]}
+   */
+  function remove(target) {
+    /** @type {bigint[]} */
     const droppedTimes = [];
     let i = 0;
     while (i < schedule.length) {
       const { time, handlers } = schedule[i];
-      if (handlers.length === 1) {
-        if (handlers[0].handler === targetHandler) {
-          schedule.splice(i, 1);
+      // Nothing prevents a particular handler from appearing more than once
+      for (let j = handlers.length - 1; j >= 0; j -= 1) {
+        if (handlers[j].handler === target) {
+          handlers.splice(j, 1);
           droppedTimes.push(time);
```

Comment:
It's current behavior that the return value can include duplicate times, but still potentially surprising. Please document in the JSDoc header, and if possible confirm that consumers behave correctly w.r.t. the possibility.